core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2434,7 +2434,8 @@ private[spark] object Utils extends Logging {
    */
   def getSparkOrYarnConfig(conf: SparkConf, key: String, default: String): String = {
     val sparkValue = conf.get(key, default)
-    if (conf.get(SparkLauncher.SPARK_MASTER, null) == "yarn") {
+    if (conf.get(SparkLauncher.SPARK_MASTER, null) == "yarn"
```

Comment:
No.

The logic you want here is the equivalent of:

```
if conf.contains(key)
  get spark conf
elif is_running_on_yarn()
  get conf from yarn
else
  return default
```

Reply:
Assuming that `--conf spark.shuffle.service.port = 7338` is configured, 7338 is displayed on the tab of the environment, but 7337 is actually used.
So my idea is get value from `SparkConf ` if  key starting with `spark.` except for `spark.hadoop.`.

Reply:
`YarnConfiguration` can only configure one `spark.shuffle.service.port` value.
We can gradually upgrade the shuffle service if get `spark.shuffle.service.port` value from `SparkConf` because we can set different values ​​for different applications.